### DISCUSSION TITLE
What version of Skia Sharp is Avalonia using now?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#13102] [Q&A] [answered]
Recently, I accidentally discovered that Skia Sharp has released a new version. (https://github.com/mono/SkiaSharp/releases/tag/v2.88.6)

In the new version, the Skia version has been upgraded to M115 (very close to the latest version of Skia, M118). 

So, I wanted to find out what version of Skia Sharp Avalonia is using now, but I couldn't find it at all. 

I wonder if the Avalonia team will simultaneously upgrade the Skia Sharp version to the latest to get better rendering performance and fix old bugs? 

Could the Avalonia team or other contributors please tell me the answer? Thank you.

**Comment by rabbitism** [comment]
I think M115 will be released with SkiaSharp 3.0, not this one. 
We have a PR pending https://github.com/AvaloniaUI/Avalonia/pull/12604

**Reply by kyuranger** [reply]
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/f25754c5-25b3-4d07-9192-09fe2cef0fb7)
I saw this description at the bottom of the SkiaSharp project page, so I thought the latest version used m115

**Reply by robloo** [reply, marked as answer]
@rabbitism is correct. M115 is for the SkiaSharp 3.0 branch (we are talking about two different branches). Avalonia is not on the 3.0 branch yet and there are breaking changes that might affect when this can be merged in.

That said, SkiaSharp will need to be updated in Avalonia to the latest 2.88.x version to get some critical bug fixes sooner rather than later.

I expect two SkiaSharp updates in Avalonia but M115 won't be in the first one.

Note that this is just my understanding and the core team may have a different plan.

**Reply by kyuranger** [reply]
All right. Thank you so much.

**Comment by dbriard** [comment]
Would be awesome if Avalonia could upgrade to SkiaSharp 3.0, it contains severals features that I am looking for.
In the meantime, are you planning to release 11.0.5 soon? There are so many bug fixes already since 11.0.4.

**Reply by timunie** [reply]
1. SkiaSharp update to 3.x is most likely a breaking change, so will take some time to figure out how to update. 
2. 11.0.5 is in the oven. We are currently checking what needs to be inside.

**Comment by dbriard** [comment]
> 1. SkiaSharp update to 3.x is most likely a breaking change, so will take some time to figure out how to update.
I know it contrains some breaking changes with 2.xx. You mean for you to rewrite code, or for end users? as in Avalonia the Skia part is not public. I am relying a lot on Skia in the app I am working on, but generally using custom draw operation and the SKCanva. I suppose that apps that rely on custom draw can update their code if necessary.

> 2. 11.0.5 is in the oven. We are currently checking what needs to be inside.
🙂

**Reply by timunie** [reply]
>  You mean for you to rewrite code, or for end users?

For the end users, so on your side. Most of our APIs are not hit by this, but some minor areas are. So we have to consider this carefully
